src/Sylius/Bundle/ApiBundle/Resources/config/routing/order_item.yml

```diff
@@ -19,6 +19,10 @@ sylius_api_order_item_create:
         _controller: sylius.controller.order_item:createAction
         _sylius:
             form: sylius_api_order_item
+            factory:
+                method: createForOrderWithId
+                arguments:
+                    type: $orderId
```

Comment:
It would be better to use an expression here, and decouple a factory from the repository. `expr:service('sylius.repository.order').find($orderId)` 

Then also, we should change a name to `createForOrder`